code/modules/vehicles/mecha/equipment/tools/work_tools.dm

```diff
@@ -11,24 +11,27 @@
 	tool_behaviour = TOOL_RETRACTOR
 	range = MECHA_MELEE
 	toolspeed = 0.8
-	var/dam_force = 20
-	var/obj/vehicle/sealed/mecha/working/ripley/cargo_holder
 	harmful = TRUE
 	mech_flags = EXOSUIT_MODULE_RIPLEY
+	///Bool for whether we beat the hell out of things we punch (and tear off their arms)
+	var/killer_clamp = FALSE
+	///How much base damage this clamp does
+	var/clamp_damage = 20
+	///Var for the chassis we are attached to, needed to access ripley contents and such
+	var/obj/vehicle/sealed/mecha/working/ripley/cargo_holder
 
-/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/can_attach(obj/vehicle/sealed/mecha/working/ripley/M)
-	if(..())
-		if(istype(M))
-			return 1
-	return 0
+/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/can_attach(obj/vehicle/sealed/mecha/M)
+	if(!.)
+		return
```

Comment:
```suggestion
/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/can_attach(obj/vehicle/sealed/mecha/M)
	. = ..()
```
Forgot to call parent here, also no need to check the parent return if I'm getting this logic correctly?

Reply:
This needs the parent call and if(!.), added